### PR TITLE
A few updates. Start on KMS section.

### PR DESCRIPTION
**Pull request by jlevy**
Several fixups based on feedback. Minimal start on #36.

### CODE REVIEW

**Comment by jlevy** [comment]
Thanks @benkehoe for the feedback! Would you care to review?

@ThanosBaskous review?

File: README.md [hunk 1]
```diff
@@ -87,11 +88,12 @@ General Information
...
+-	🔹**EC2 vs. other services:** Most users of AWS are most familiar with [EC2](#ec2), AWS’ flagship virtual server product, and possibly a few others like S3 and ELBs. But AWS products now extend far beyond basic IaaS, and often companies do not properly understand or appreciate all the many AWS services and how they can be applied, due to the [sharply growing](#which-services-to-use) number of services, their novelty and complexity, branding confusion, and fear of ⛓lock-in to proprietary AWS technology. Although a bit daunting, it’s important for technical decision-makers in companies to understand the breadth of the AWS services and make informed decisions. (We hope this guide will help.)
+-	🚪**AWS vs. other cloud providers:** While AWS is the dominant IaaS provider (31% market share in [this 2016 estimate](https://www.srgresearch.com/articles/aws-remains-dominant-despite-microsoft-and-google-growth-surges)), there is significant of competition and alternatives that are better suited to some companies:
```
Comment (ThanosBaskous) [inline README.md]: "there is significant of competition" -> "there is significant competition"

File: README.md [hunk 2]
```diff
@@ -849,10 +857,12 @@ API Gateway
 -	📒 [Homepage](https://aws.amazon.com/api-gateway/) ∙ [Developer guide](http://docs.aws.amazon.com/apigateway/latest/developerguide/) ∙ [FAQ](https://aws.amazon.com/api-gateway/faqs/) ∙ [Pricing](https://aws.amazon.com/api-gateway/pricing/)
 -	**API Gateway** provides a scalable, secured front-end for service APIs, and can work with Lambda, Elastic Beanstalk, or regular EC2 services.
 -	It allows “serverless” deployment of applications built with Lambda.
+-	🔸Switching over deployments after upgrades can be tricky. There are no built-in mechanisms to have a single domain name migrate fro one API gateway to another one. So it may be necessary to build an additional layer in front (even another API Gateway) to allow smooth migration from one deployment to another.
```
Comment (ThanosBaskous) [inline README.md]: "migrate fro one" -> "migrate from one"

**Comment by ThanosBaskous** [comment]
Great! 👍
lgtm @jlevy (minus a couple of typos).

**Pull request merged**